pinot-segment-local/src/main/java/org/apache/pinot/segment/local/recordtransformer/ExpressionTransformer.java

```diff
@@ -125,8 +127,15 @@ public GenericRow transform(GenericRow record) {
       // Skip transformation if column value already exist.
       // NOTE: column value might already exist for OFFLINE data
       if (record.getValue(column) == null) {
-        Object result = transformFunctionEvaluator.evaluate(record);
-        record.putValue(column, result);
+        if (_continueOnError) {
+          try {
+            record.putValue(column, transformFunctionEvaluator.evaluate(record));
+          } catch (Exception e) {
+            record.putValue(column, null);
```

Comment:
This `putValue()` is redundant. Also we should consider logging something (similar to `DataTypeTransformer`)